src/HDF5.jl

```diff
@@ -74,25 +74,25 @@ cset(::Type{UTF8Char}) = H5T_CSET_UTF8
 cset(::Type{ASCIIChar}) = H5T_CSET_ASCII
 
 ## Conversion between Julia types and HDF5 atomic types
-hdf5_type_id(::Type{Bool})       = H5T_NATIVE_B8
-hdf5_type_id(::Type{Int8})       = H5T_NATIVE_INT8
-hdf5_type_id(::Type{UInt8})      = H5T_NATIVE_UINT8
-hdf5_type_id(::Type{Int16})      = H5T_NATIVE_INT16
-hdf5_type_id(::Type{UInt16})     = H5T_NATIVE_UINT16
-hdf5_type_id(::Type{Int32})      = H5T_NATIVE_INT32
-hdf5_type_id(::Type{UInt32})     = H5T_NATIVE_UINT32
-hdf5_type_id(::Type{Int64})      = H5T_NATIVE_INT64
-hdf5_type_id(::Type{UInt64})     = H5T_NATIVE_UINT64
-hdf5_type_id(::Type{Float32})    = H5T_NATIVE_FLOAT
-hdf5_type_id(::Type{Float64})    = H5T_NATIVE_DOUBLE
-hdf5_type_id(::Type{Reference})  = H5T_STD_REF_OBJ
-
-hdf5_type_id(::Type{S}) where {S<:AbstractString}  = H5T_C_S1
-hdf5_type_id(::Type{C}) where {C<:CharType}        = H5T_C_S1
-
-const HDF5BitsKind = Union{Bool, Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64, Float32, Float64}
-const HDF5Scalar = Union{HDF5BitsKind, Reference}
-const ScalarOrString = Union{HDF5Scalar, String}
+hdf5_type_id(::Type{Bool})      = H5T_NATIVE_B8
+hdf5_type_id(::Type{Int8})      = H5T_NATIVE_INT8
+hdf5_type_id(::Type{UInt8})     = H5T_NATIVE_UINT8
+hdf5_type_id(::Type{Int16})     = H5T_NATIVE_INT16
+hdf5_type_id(::Type{UInt16})    = H5T_NATIVE_UINT16
+hdf5_type_id(::Type{Int32})     = H5T_NATIVE_INT32
+hdf5_type_id(::Type{UInt32})    = H5T_NATIVE_UINT32
+hdf5_type_id(::Type{Int64})     = H5T_NATIVE_INT64
+hdf5_type_id(::Type{UInt64})    = H5T_NATIVE_UINT64
+hdf5_type_id(::Type{Float32})   = H5T_NATIVE_FLOAT
+hdf5_type_id(::Type{Float64})   = H5T_NATIVE_DOUBLE
+hdf5_type_id(::Type{Reference}) = H5T_STD_REF_OBJ
+
+hdf5_type_id(::Type{S}) where {S<:AbstractString} = H5T_C_S1
+hdf5_type_id(::Type{C}) where {C<:AbstractChar} = H5T_C_S1
```

Comment:
I just noticed — `CharType <: AbstractString`, so this method shouldn't actually be necessary since the previous one covers all cases.